### PR TITLE
Add a global setting for Ctrl-F focus (improve #14947)

### PR DESCRIPTION
**Pull request by etiennewan**
This PR adds a global setting to enable or disable the global search bar focus with Ctrl-F keyboard shortcut, introduced in #14947.
image

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [x] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [x] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [ ] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by electrocret** [comment]
Heh, I wondered how long until someone made a PR like this. I'm still getting used to it myself. 
I was kind of hoping whoever made the disabling of this feature a User setting instead of a Global setting. 

Code looks good to me, however I'll let another maintainer decide whether to push for User vs Global.

**Comment by etiennewan** [comment]
I was not sure how to implement a user setting so I went with a global setting for the moment. Do you have any code resource for a user setting implementation?

**Comment by electrocret** [comment]
> I was not sure how to implement a user setting so I went with a global setting for the moment. Do you have any code resource for a user setting implementation?

Heh, I'm not sure either. I believe it's not as streamlined as global settings.

**Comment by electrocret** [comment]
Did some poking in the code, using "Add schedule notes to devices notes" preference as an example, and came up with this:

Preference is Defined:
https://github.com/librenms/librenms/blob/927446db7bdc003ad44f6d69a34494115b131fa6/app/Http/Controllers/UserPreferencesController.php#L69

Toggle in GUI:
https://github.com/librenms/librenms/blob/927446db7bdc003ad44f6d69a34494115b131fa6/resources/views/user/preferences.blade.php#L96

Use of Preference:
https://github.com/librenms/librenms/blob/927446db7bdc003ad44f6d69a34494115b131fa6/includes/html/forms/schedule-maintenance.inc.php#L154

**Comment by etiennewan** [comment]
Thanks

> Did some poking in the code, using "Add schedule notes to devices notes" preference as an example, and came up with this:
> 
> Preference is Defined in Controller:
> 
> https://github.com/librenms/librenms/blob/927446db7bdc003ad44f6d69a34494115b131fa6/app/Http/Controllers/UserPreferencesController.php#L69
> 
> https://github.com/librenms/librenms/blob/927446db7bdc003ad44f6d69a34494115b131fa6/app/Http/Controllers/UserPreferencesController.php#L104
> 
> Toggle in GUI:
> 
> https://github.com/librenms/librenms/blob/927446db7bdc003ad44f6d69a34494115b131fa6/resources/views/user/preferences.blade.php#L96
> 
> https://github.com/librenms/librenms/blob/927446db7bdc003ad44f6d69a34494115b131fa6/resources/views/user/preferences.blade.php#L194
> 
> Use of Preference:
> 
> https://github.com/librenms/librenms/blob/927446db7bdc003ad44f6d69a34494115b131fa6/includes/html/forms/schedule-maintenance.inc.php#L154

Thanks a lot for your help and insights! This new commit moves this setting to user settings : 
image

**Review by electrocret — COMMENTED** [review]
LGTM. I'll let someone who's more familiar with Laravel verify & approve/merge.

**Comment by etiennewan** [comment]
Spotted my mistake in wrapping the javascript code in menu.blade.php

**Review by electrocret — APPROVED** [review]

**Pull request merged**
